proto/umee/leverage/v1/events.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto3";
+package umee.leverage.v1;
+
+import "cosmos/base/v1beta1/coin.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/umee-network/umee/v3/x/leverage/types";
+
+option (gogoproto.goproto_getters_all) = false;
+
+// EventSupply is emitted on Msg/Supply
+message EventSupply {
+  // Liquidity supplier bech32 address.
+  string supplier = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // Liquidity provided to the module.
+  cosmos.base.v1beta1.Coin supplied = 2 [(gogoproto.nullable) = false];
+  // uAsset received by the supplier in exchange for the
+  // provided liquidity.
+  cosmos.base.v1beta1.Coin uasset = 3 [(gogoproto.nullable) = false];
+}
+
+// EventWithdraw is emitted on Msg/Withdraw
+message EventWithdraw {
+  // Liquidity supplier bech32 address.
+  string supplier = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // uAsset sent to the module in exchange for the underlying asset.
+  cosmos.base.v1beta1.Coin uasset = 2 [(gogoproto.nullable) = false];
+  // Liquidity received by the supplier.
+  cosmos.base.v1beta1.Coin asset = 3 [(gogoproto.nullable) = false];
+}
+
+// EventCollaterize is emitted on Msg/Collaterize
+message EventCollaterize {
+  // Borrower bech32 address.
+  string borrower = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // uAsset provided as a collateral.
+  cosmos.base.v1beta1.Coin uasset = 2 [(gogoproto.nullable) = false];
+}
+
+// EventDecollaterize is emitted on Msg/Decollateralize
+message EventDecollaterize {
+  // Borrower bech32 address.
+  string borrower = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // uAsset removed from collateral.
+  cosmos.base.v1beta1.Coin uasset = 2 [(gogoproto.nullable) = false];
+}
+
+// EventBorrow is emitted on Msg/Borrow
+message EventBorrow {
+  // Borrower bech32 address.
+  string borrower = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // Asset borrowed.
+  cosmos.base.v1beta1.Coin asset = 2 [(gogoproto.nullable) = false];
+}
+
+// EventRepay is emitted on Msg/Repay
+message EventRepay {
+  // Borrower bech32 address.
+  string borrower = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // Asset repaid
+  cosmos.base.v1beta1.Coin repaid = 2 [(gogoproto.nullable) = false];
+}
+
+// EventLiquidate is emitted on Msg/Liquidate
+message EventLiquidate {
+  // Liquidator bech32 address.
+  string liquidator = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // Borrower bech32 address.
+  string borrower = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // Assets liquidated from the borrower
+  cosmos.base.v1beta1.Coin liquidated = 3 [(gogoproto.nullable) = false];
```

Comment:
```suggestion
  // Assets repaid by the liquidator
  cosmos.base.v1beta1.Coin repaid = 4 [(gogoproto.nullable) = false];
  // Collateral uTokens liquidated from the borrower
  cosmos.base.v1beta1.Coin liquidated = 4 [(gogoproto.nullable) = false];
  // Rewards transferred to the liquidator
  cosmos.base.v1beta1.Coin reward = 5 [(gogoproto.nullable) = false];
```
These three fields cannot be derived from the Msg (but are present in the MsgResponse).

Reply:
Do you suggest to remove them? I thought to keep them in case someone will make cross-module liquidation call.

Reply:
We should have all three instead of just one - that would be consistent with the other messages, where response fields are present in the event.

Reply:
I was explaining that yesterday during the sync. It doesn't make sense to duplicate that.

Reply:
In that case `liquidated` field could be eliminated as well, since it's also in the response.

Or, if we're making the event from fields that the borrower needs to know (since they're not the sender) we can use `repaid` and `liquidated` but omit `reward`.

Reply:
`liquidated` is the essential attribute here. This is the one informing what was liquidated and from who.